astropy/cosmology/flrw/tests/test_base.py

```diff
@@ -68,15 +68,17 @@ def test_H0(self, cosmo_cls, cosmo):
         unit = u.Unit("km/(s Mpc)")
 
         # on the class
-        assert isinstance(cosmo_cls.H0, Parameter)
-        assert "Hubble constant" in cosmo_cls.H0.__doc__
-        assert cosmo_cls.H0.unit == unit
+        H0 = cosmo_cls.parameters["H0"]
+        assert isinstance(H0, Parameter)
+        assert "Hubble constant" in H0.__doc__
+        assert H0.unit == unit
+        assert H0.default is NotImplemented
```

Comment:
It's because I wanted a sentinel value and `None` wan't available (`None` is often used but can't be here because it's a valid input for some Parameter). We could spin our own, using the conventional if somewhat inconvenient code pattern

```python
class Sentinel(Enum):
    MISSING = default()

MISSING = Sentinel.MISSING  
```

My issue with this is that `eval(repr())` fails, while it works with `NotImplemented`.
Other than that it's fine. An easy swap to make.

Reply:
in `inspect` there's `_empty` but that's not normally publicly visible.

Reply:
[Python needs Sentinels](https://peps.python.org/pep-0661/)

Reply:
See #15400